### PR TITLE
Unit Testing Task

### PR DESCRIPTION
**Pull request by ipelekhan**
- [x] Finish previous task (add `@Output` and simple routing)
- [x] Make use of `ngModule`
- [x] Add service with Observable
- [x] Add `@testing-library/angular` and write a test with it
- [x] Increase coverage
- [x] Fix broken unit tests
- [x] Add more tests!

### CODE REVIEW

File: src/app/app.component.spec.ts [hunk 1]
```diff
@@ -1,41 +1,16 @@
...
-        CoursesPageComponent,
-        CourseListComponent,
-        CourseItemComponent,
-        FooterComponent,
-        SearchBarComponent,
-        ButtonComponent,
-        LabelWithIconComponent,
-        ButtonWithIconComponent,
-      ],
+      imports: [RouterTestingModule, LayoutModule, CoursesModule],
```
Comment (dragonflypl) [inline src/app/app.component.spec.ts:12]: Nice job

File: src/app/courses/course-item/course-item.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    'How to make your guests mad.',
+    123
+  );
+
+  it('should create', async () => {
+    await render(CourseItemComponent, {
+      imports: [CoursesModule],
+      componentProperties: {
+        course: testCourse,
+        deleteCourseRequest: {
```
Comment (dragonflypl) [inline src/app/courses/course-item/course-item.component.spec.ts:20]: you don't need to pass it

**Review by dragonflypl — CHANGES_REQUESTED** [review]
Good job.

Next time:
- please do changes not related to homework in separate commits (like modules refactoring). It is hard to check howework when there is this amoun of changes
- I didn't see `Test host testing` case. Please add it (e.g. create test for pipes with test host component.

**Comment by ipelekhan** [comment]
Hm, not sure what to fix. On my both machine it's green:
image

File: src/app/courses/course-item/course-item.component.spec.ts (hunk 2 above)
Comment (ipelekhan) [reply src/app/courses/course-item/course-item.component.spec.ts:20]: Could you please elaborate?

**Comment by ipelekhan** [comment]
@dragonflypl I am going to merge this PR, could you please mark `Task 3` as resolved at learn.epam.com?

**Pull request merged**